### DISCUSSION TITLE
How do you enable debug logging using the 1.5.3 client?

### DISCUSSION THREAD

**Discussion by mindwerxkmg** [edenhill/librdkafka#3229] [Q&A]
I cannot get debug logging to work with the 1.5.3 client.  The older client we were using provided an OnLog event which does not exist in the 1.5.3 version.

Any assistance would be helpful!

**Comment by rolandyoung** [comment]
When writing a C program using librdkafka, you provide a logger callback function via `rd_kafka_conf_set_log_cb`. In C++, you subclass `EventCb`.
What language is your application written in?  (Maybe Go or C#?) 

librdkafka is the C/C++ library underlying a number of [language bindings](https://github.com/edenhill/librdkafka#language-bindings) and it sounds as if you are describing a feature in one of those bindings. You might want to ask your question in the support forum for the binding you are using.

**Reply by mindwerxkmg** [reply]
Using C#.  Found SetLogHandler.  Thanks!

**Comment by JSoet** [comment]
When using C++ `EventCb`, is that the equivalent of the c `rd_kafka_conf_set_log_cb`? Everything that would normally be passed to the c log callback would be passed to the event callback?

**Reply by edenhill** [reply]
Yes
